packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -10,6 +10,33 @@ import 'package:serverpod_serialization/serverpod_serialization.dart';
 typedef constructor<T> = T Function(
     dynamic jsonSerialization, SerializationManager serializationManager);
 
+/// Exception thrown when no deserialization type was found during
+/// protocol deserialization
+class DeserializationTypeNotFoundException implements Exception {
+  /// The exception message that was thrown.
+  final String message;
+
+  /// The type that was not found.
+  final Type? type;
+
+  /// Creates a new [DeserializationTypeNotFoundException].
+  DeserializationTypeNotFoundException(
+    this.message, {
+    this.type,
+  });
+
+  /// Creates a new [DeserializationTypeNotFoundException] with the given type.
+  factory DeserializationTypeNotFoundException.typeError(Type t) {
+    return DeserializationTypeNotFoundException(
+      'No deserialization found for type $t',
+      type: t,
+    );
+  }
```

Comment:
Suggestion: Construct it in the message directly since we always want to provide the same message for this type.

`final String message = 'No deserialization found for type $t';`

That way the user of the exception doesn't need to remember to call typeError to conform to the interface. And since this has a specific meaning in the serverpod framework I think it is fine we decide what the error message is.